### PR TITLE
Fix infinite spinner when signing typed messages

### LINKED ISSUE #19826: [Bug]: Metamask extension freezes when eth _sign_typedData_ v4 is called in recent version
### Describe the bug

When clicking on a link that worked prior (within my app and others) which prompts an `eth_sign_typedData_v4` the extension freezes and I cannot clear it our unless I restart my browser.
This issue is causing log in to with the method to be impossible

image

Code used:
```
const signature = await this._web3.givenProvider.request({
      method: 'eth_signTypedData_v4',
      params: [account.address, JSON.stringify(typedMessage)],
    });
 ```

### Steps to reproduce

1. Attempt to call eth_SignedData_v4

### Error messages or log output

```shell
No console error messages
```

### Version

10.32

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by matthewwalsh0**
## Explanation

A race condition in the `@metamask/message-manager` package was causing the UI to redirect in a loop and display a spinner infinitely.

This fix adopts the updated core package.

Fixes: #19826 

## Manual Testing Steps

- Go to the test dApp.
- Go to `Settings > Experimental` and enable security alerts.
- Open the developer console for the background page and apply network throttling.
- Sign a typed message.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,6 +101,7 @@
     "@babel/core": "patch:@babel/core@npm%3A7.21.5#./.yarn/patches/@babel-core-npm-7.21.5-c72c337956.patch",
     "@babel/runtime": "patch:@babel/runtime@npm%3A7.18.9#./.yarn/patches/@babel-runtime-npm-7.18.9-28ca6b5f61.patch",
     "@metamask/approval-controller": "^3.3.0",
+    "@metamask/message-manager": "^7.0.2",
```
Comment (FrederikBolding) [inline package.json:104]: We don't really need a resolution when all other deps are on v7 do we? 🤔 

Should be enough with a regular patch bump.

Comment (matthewwalsh0) [reply package.json:104]: Good point, removed the resolution but updated the lock file.

Comment (FrederikBolding) [reply package.json:104]: Looks like you may want to run `yarn dedupe` too

**Review by vinistevam — APPROVED** [review]

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Comment by danjm** [comment]
I'm going to merge because I want to get this into a hotfix release. But if you wanted to hold off on merging, just let me know and we can revert if necessary @matthewwalsh0

**Pull request merged**